src/main/java/Duke.java

```diff
@@ -1,10 +1,377 @@
+            hasSaved = false;
+        }
+        return hasSaved;
+    }
+
+    private static boolean writeTaskToFile(FileWriter fw, TaskList t) {
+        boolean hasSaved = true;
+        if (t instanceof Event) {
+            Event temp = (Event) t;
+            try {
+                fw.write(temp.getTaskToPrintInFile());
+            } catch (java.io.IOException e) {
+                hasSaved = false;
+            }
+
+        } else if (t instanceof Deadline) {
+            Deadline temp = (Deadline) t;
+            try {
+                fw.write(temp.getTaskToPrintInFile());
+            } catch (java.io.IOException e) {
+                hasSaved = false;
+            }
+
+        } else if (t instanceof Todo) {
+            Todo temp = (Todo) t;
+            try {
+                fw.write(temp.getTaskToPrintInFile());
+            } catch (java.io.IOException e) {
+                hasSaved = false;
+            }
+        }
+        return hasSaved;
+    }
+
+
+    private static void processCommands() {
+        String line;
+        Scanner in = new Scanner(System.in);
+        boolean hasToContinue = true;
+
+        dukeList.printHelp();
+        while (hasToContinue) {
+            line = in.nextLine();
+            hasToContinue = selectCommand(line);
+        }
+
+    }
+
+    private static boolean selectCommand(String line) {
+        switch (line) {
+        case "bye":
+            exit();
+            return false;
+        case "list":
+            try {
+                printAllLists();
+            } catch (IllegalListException e) {
+                dukeList.printEmptyList();
+            }
+            break;
+        case "help":
+            dukeList.printHelp();
+            break;
+
+        default:
+            handleAmendList(line);
+            break;
+        }
+        return true;
+    }
+
+    private static void handleAmendList(String line) {
+        try {
+            amendList(line);
+        } catch (IllegalCommandException e) {
+            dukeList.printCommandDoesNotExist();
+        } catch (IllegalTaskException e) {
+            dukeList.printInvalidTaskPhrase();
+        } catch (IllegalTaskRedoException e) {
+            printTaskAlreadyCompletedPhrase();
+        }
+        dukeList.printDottedLines();
+    }
+
+    private static void printAllLists() throws IllegalListException {
+        int i = 1;
+        if (tasks.size() == 0) {
+            throw new IllegalListException();
+        }
+        dukeList.printListName();
+        for (TaskList t : tasks) {
+            System.out.print(i + ". ");
+            t.printTask();
+            i++;
+        }
+        printNumberOfTasksLeft();
+        dukeList.printDottedLines();
+    }
+
+    private static void amendList(String line) throws IllegalCommandException, IllegalTaskException,
+            IllegalTaskRedoException {
+        String[] sentence = line.split(" ");
+        if (sentence.length < 2) {
```

Comment:
Avoid magic numbers -- consider declaring this as a constant with a descriptive name that explains what this 2 mean